Source/JavaScriptCore/dfg/DFGObjectAllocationSinkingPhase.cpp

```diff
@@ -2058,6 +2207,8 @@ class ObjectAllocationSinkingPhase : public Phase {
                 bool desiredNextExitOK = node->origin.exitOK && !clobbersExitState(m_graph, node);
 
                 bool doLower = false;
+                m_nodeIndex = nodeIndex;
```

Comment:
Let's not have this member variable. `handleNode` is used in multiple places but only this place is setting m_nodeIndex, that sounds dangerous. We should pass nodeIndex to handleNode if it is necessary.